cli_tools/google-osconfig-agent/ospatch/patch_run.go

```diff
@@ -352,54 +338,80 @@ func ackPatch(ctx context.Context, patchJobName string) {
 		return
 	}
 
-	// Notify the server if we haven't yet. If we've already been notified about this Job,
+	// Ack if we haven't yet. If we've already been notified about this Job,
 	// the server may have inadvertantly notified us twice (at least once deliver) so we
 	// can ignore it.
 	if currentPatchJob == nil || currentPatchJob.Job.PatchJob != patchJobName {
-		res, err := reportPatchDetails(ctx, patchJobName, osconfigpb.Instance_NOTIFIED, 0, "")
-		if err != nil {
+		j := &patchJob{&osconfigpb.ReportPatchJobInstanceDetailsResponse{PatchJob: patchJobName}}
+		pr := &patchRun{ctx: ctx, Job: j}
+		if err := pr.createClient(); err != nil {
+			logger.Errorf("Error creating osconfig client: %v", err)
+		}
+		if err := pr.reportPatchDetails(osconfigpb.Instance_ACKED, 0, ""); err != nil {
 			logger.Errorf("reportPatchDetails Error: %v", err)
+			pr.close()
 			return
 		}
-		tasker.Enqueue("Run patch", func() {
-			patchRunner(ctx, &patchRun{Job: &patchJob{res}})
-		})
+		pr.PatchStep = acked
```

Comment:
So if the process crashes before the patch job is started, it will ack again. Isn't that what we want to avoid? Should this be saved before returning?

Reply:
This is fixed in my 'state' PR